editor/gui/editor_run_bar.cpp

```diff
@@ -356,14 +360,39 @@ HBoxContainer *EditorRunBar::get_buttons_container() {
 	return main_hbox;
 }
 
+void EditorRunBar::_instance_starting(int p_idx, List<String> &r_arguments) {
+	singleton->_instance_starting_internal(p_idx, r_arguments);
+}
+
+void EditorRunBar::_instance_starting_internal(int p_index, List<String> &r_arguments) {
+	Array arguments;
+
+	// We need to convert the an Array so it can be passed as parameter in a signal.
+	for (const String &arg : r_arguments) {
+		arguments.push_back(arg);
+	}
+
+	emit_signal(SNAME("instance_starting"), p_index, arguments);
+
+	// Copy back to a List.
+	r_arguments.clear();
+	for (const Variant &arg : arguments) {
+		r_arguments.push_back(arg);
+	}
+}
```

Comment:
It would be nice to avoid this if `arguments` is not modified.

Reply:
Done! By using the callback directly, I was able to fix this issue.